clang/lib/Sema/SemaDecl.cpp

```diff
@@ -9759,7 +9759,7 @@ Sema::ActOnFunctionDeclarator(Scope *S, Declarator &D, DeclContext *DC,
   SmallVector<TemplateParameterList *, 4> TemplateParamLists;
   llvm::append_range(TemplateParamLists, TemplateParamListsRef);
   if (TemplateParameterList *Invented = D.getInventedTemplateParameterList()) {
-    if (!TemplateParamLists.empty() &&
+    if (!TemplateParamLists.empty() && TemplateParamLists.back()->size() &&
```

Comment:
```suggestion
    if (!TemplateParamLists.empty() && !TemplateParamLists.back()->empty() &&
```

Reply:
Ah, urgh, i hate 'sorta vector' types.  Yeah, i think 'empty' makes a lot of sense, it improves the readability a ton in a situation like this.

Reply:
Added. I'll open a separate PR that converts other uses of `size()` to use `empty()` where appropriate